index.js

```diff
@@ -0,0 +1,273 @@
+        if (!response.ok) {
+            throw new Error(`HTTP error! Status: ${response.status}`);
+        }
+        sunData = yield response.json();
+        console.log(sunData);
+        displaySunTimes(sunData);
+    }
+    catch (error) {
+        console.error("An error:", error);
+    }
+});
+//----------------------------------
+// Sunrise/sunset function
+//----------------------------------
+const displaySunTimes = (sunData) => {
+    const sunrise = sunData.results.sunrise;
+    const sunset = sunData.results.sunset;
+    sunriseSunsetContainer.innerHTML = `
+  <div class="top-info">
+  <p id="topSunrise">Sunrise</p>
+  <p id="sunriseTime">${sunrise}</p>
+  </div>
+  
+  <div class="top-info">
+  <p id="topSunset">sunset</p>
+  <p id="sunsetTime">${sunset}</p>
+  </div>
+  `;
+};
+fetchSunriseSunsetData();
+//Mapping codes and conditions:
+const weatherSymbol = (code) => {
+    const mapping = {
+        1: "Clear sky",
+        2: "Nearly clear",
+        3: "Variable clouds",
+        4: "Half cloudy sky",
+        5: "Cloudy",
+        6: "Overcast",
+        7: "Fog",
+        8: "Light rain",
+        9: "Moderate rain",
+        10: "Heavy showers",
+        11: "Thunderstorm",
+        12: "Light sleet",
+        13: "Moderate sleet",
+        14: "Heavy sleet",
+        15: "Light snow",
+        16: "Moderate snow",
+        17: "Heavy snow",
+        18: "Light rain",
+        19: "Moderate rain",
+        20: "Heavy rain",
+        21: "Thunder",
+        22: "Light sleet",
+        23: "Moderate sleet",
+        24: "Heavy sleet",
+        25: "Light snow",
+        26: "Moderate snow",
+        27: "Heavy snow"
+    };
+    return mapping[Math.round(code)] || "Unknown";
+};
+//----------------------------------
+// Show todays forecast function
+//----------------------------------
+const todayForecast = () => {
+    const current = data.timeSeries[0].data;
+    const airTemp = current.air_temperature;
+    const conditionCode = current.symbol_code;
+    const conditionLabel = weatherSymbol(conditionCode);
+    const rainData = current.probability_of_precipitation;
+    const condition = weatherSymbol(data.timeSeries[0].data.symbol_code);
+    todayWeather = {
+        condition,
+        airTemp,
+        rainData
+    };
+    weatherContainer.innerHTML = `
+  <div class="top-info">
+  <p id="topCondition">${todayWeather.condition}</p>
+  <hr>
+  <p id="topTemp">${todayWeather.airTemp}&deg;</p>
+  </div>
+  `;
+    rainContainer.innerHTML = `
+  <div class="top-info">
+  <p id="topRainData">Probability of precipitation: ${todayWeather.rainData}%</p>
+  </div>`;
+    showMessage(todayWeather, adviceContainer, weeklyTempContainer);
+};
+//----------------------------------
+// Function for showing advice message, symbol and diffrent styling depending on weather
+//----------------------------------
+const showMessage = (data, adviceContainer, weeklyTempContainer) => {
```

Comment:
Nice job, one way of refactoring this to make it a bit less "cluttery" with the styling inside of the showMessage function is to set the styling, colors like you did with the mapping of symbols and weather descriptions. Or to add a class to the html element and deal with the styling in css